spacy/errors.py

```diff
@@ -960,6 +960,7 @@ class Errors(metaclass=ErrorsWithCodes):
     E4003 = ("Training examples for distillation must have the exact same tokens in the "
              "reference and predicted docs.")
     E4004 = ("Backprop is not supported when is_train is not set.")
+    E4005 = ("Expected `entity_id` to be of type {should_type}, but is of type {is_type}.")
```

Comment:
I wouldn't use "should" or "is" here as part of the variable names, but instead something like "exp" and "value"/"found" to be more in line with other error messages (not that we're that consistent, but this feels off to me)